test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs

```diff
@@ -607,6 +607,199 @@ public void TestServerChanges()
+        messageFormatter: ""Tag {0} is not valid."",
+        openingTagOnly: true
+    }
+]
+}");
+
+            FileCollection files = new FileCollection(Directory.GetCurrentDirectory());
+            files.Add(DocumentType.Article, new[] { tocFile, conceptualFile, conceptualFile2, conceptualFile3, conceptualFile4 });
+            files.Add(DocumentType.Article, new[] { "TestData/System.Console.csyml", "TestData/System.ConsoleColor.csyml" }, "TestData/", null);
+            files.Add(DocumentType.Resource, new[] { resourceFile });
+            #endregion
+
+            Init("IncrementalBuild.TestFilesAddRemoveFromDocfx");
+            var outputFolderFirst = Path.Combine(outputFolder, "IncrementalBuild.TestFilesAddRemoveFromDocfx");
+            var outputFolderForIncremental = Path.Combine(outputFolder, "IncrementalBuild.TestFilesAddRemoveFromDocfx.Second");
+            var outputFolderForCompare = Path.Combine(outputFolder, "IncrementalBuild.TestFilesAddRemoveFromDocfx.Second.ForceBuild");
+            try
+            {
+                using (new LoggerPhaseScope("first-IncrementalBuild.TestFilesAddRemoveFromDocfx"))
+                {
+                    BuildDocument(
+                        files,
+                        inputFolder,
+                        outputFolderFirst,
+                        new Dictionary<string, object>
+                        {
+                            ["meta"] = "Hello world!",
+                        },
+                        templateFolder: templateFolder,
+                        intermediateFolder: intermediateFolder);
+
+                }
+                files.Add(DocumentType.Article, new[] { conceptualFile5 });
+                files.RemoveAll(f => f.File == conceptualFile2.ToNormalizedPath());
+                using (new LoggerPhaseScope("second-IncrementalBuild.TestFilesAddRemoveFromDocfx"))
+                {
+                    
+                    BuildDocument(
+                        files,
+                        inputFolder,
+                        outputFolderForIncremental,
+                        new Dictionary<string, object>
+                        {
+                            ["meta"] = "Hello world!",
+                        },
+                        templateFolder: templateFolder,
+                        intermediateFolder: intermediateFolder,
+                        changes: new Dictionary<string, ChangeKindWithDependency>
+                        {
+                            { ((RelativePath)conceptualFile3).GetPathFromWorkingFolder(), ChangeKindWithDependency.Updated },
+                            { ((RelativePath)conceptualFile).GetPathFromWorkingFolder(), ChangeKindWithDependency.None },
+                            { ((RelativePath)conceptualFile4).GetPathFromWorkingFolder(), ChangeKindWithDependency.None },
+                            { ((RelativePath)conceptualFile5).GetPathFromWorkingFolder(), ChangeKindWithDependency.None },
```

Comment:
for server, they only track whether the file changes.
the case is to test the scenario that file doesn't change, but is added/removed from docfx. so the changelist doesn't contain the files.